src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs

```diff
@@ -487,6 +487,12 @@ public void RelayTransaction(uint256 hash)
             IEnumerable<MempoolBehavior> behaviours = peers.Select(s => s.Behavior<MempoolBehavior>());
             foreach (MempoolBehavior mempoolBehavior in behaviours)
             {
+                if (mempoolBehavior?.AttachedPeer == null)
```

Comment:
Do we have anything that cleans out old behaviours when they are no longer attached to anything?

Reply:
ah sorry this is per behaviour so ok its correct